modules/consensus/accept.go

```diff
@@ -179,9 +186,22 @@ func (cs *ConsensusSet) AcceptBlock(b types.Block) error {
 	if len(changeEntry.AppliedBlocks) > 0 {
 		cs.readlockUpdateSubscribers(changeEntry)
 	}
+	return nil
+}
 
+// AcceptBlock will try to add a block to the consensus set. If the block does
+// not extend the longest currently known chain, an error is returned but the
+// block is still kept in memory. If the block extends a fork such that the
+// fork becomes the longest currently known chain, the consensus set will
+// reorganize itself to recognize the new longest fork. If a block is accepted
+// without error, it will be relayed to all connected peers. This function
+// should only be called for new blocks.
+func (cs *ConsensusSet) AcceptBlock(b types.Block) error {
+	err := cs.managedAcceptBlock(b)
+	if err != nil {
+		return err
+	}
 	// Broadcast the new block to all peers.
 	go cs.gateway.Broadcast("RelayBlock", b)
```

Comment:
if you are receiving multiple blocks at once, and your calls to AcceptBlock are happening faster than your calls to Broadcast, you'll end up with multiple broadcasts waiting before any have finished. Since it's happening concurrently, you have no guarantee about which order the broadcasts are happening in.